### PR TITLE
[AIRFLOW-2550] Implements API endpoint to list DAG runs

### PR DESCRIPTION
**Pull request by verdan**
Make sure you have checked _all_ steps below.

### JIRA
- [x] My PR addresses the following [Airflow JIRA](https://issues.apache.org/jira/browse/AIRFLOW-2550/) issues and references them in the PR title.

### Description
- [x] Here are some details about my PR, including screenshots of any UI changes:
Implements a new endpoint to get a list of all DAG runs. It also supports a query parameter to filter out the DAG runs based on the "state" i.e., `?state=queued|running|success...`. 
*Usage: api/experimental/dags/<string:dag_id>/dag_runs?state=<queued|running|success...>*

### Tests
- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:
`tests/www_rbac/api/experimental/test_dag_runs_endpoint.py:TestDagRunsEndpoint`

### Commits
- [x] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

### Documentation
- [x] In case of new functionality, my PR adds documentation that describes how to use it.
    - When adding new operators/hooks/sensors, the autoclass documentation generation needs to be added.

### Code Quality
- [x] Passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

**Comment by verdan** [comment]
@sekikn @jgao54 @feng-tao PTAL

File: airflow/www_rbac/api/experimental/endpoints.py [hunk 1]
```diff
@@ -115,6 +138,7 @@ def task_info(dag_id, task_id):
     return jsonify(fields)
 
 
+# ToDo: Shouldn't this be a PUT method?
```
Comment (Fokko) [inline airflow/www_rbac/api/experimental/endpoints.py:141]: Please create a JIRA for this.

**Review by Fokko — APPROVED** [review]
LGTM

**Comment by rcorre** [comment]
Thanks for the implementation! I noticed this was only added to `/www_rbac/api/experimental/endpoints.py` and not `www/api/experimental/endpoints.py`. What's the difference? When I run the `airflow webserver` I seem to get the endpoints from the latter. Should this endpoint be added to both?

**Comment by verdan** [comment]
@rcorre it was implemented using the new Flask App Builder (FAB) code base. www_rbac is the new FAB's UI code that can be turned on using a feature flag in configurations. 
```
# Use FAB-based webserver with RBAC feature
rbac = False
```

**Comment by rcorre** [comment]
@verdan got it, thanks! Is www_rbac intended to completely replace www?

**Comment by rcorre** [comment]
Any idea what release this might make it into? I'm having a hard time understanding the branching/tagging scheme here.

```
$ git describe bb0d9ee1
1.6.1-2949-gbb0d9ee1
```

Which seems weird, since the latest is 1.9.

**Comment by mhousley** [comment]
This commit is in the v1-10-test branch, so will be in the upcoming release. Looks like we're not using annotated tags, so you have to use the --tags option for light tags.
```
git describe bb0d9ee1 --tags
airbnb_1.7.1rc10-1829-gbb0d9ee1
```
In terms of branching, the release tags generally don't go on master, but on the prerelease test and stable branches. Running
```
git checkout v1-9-test
git describe --tags
```
yields `1.9.0rc1-20-g1b53882a`.
There isn't yet a 1.10 tag on branch v1-10-test.

Thanks far asking about this - didn't know the difference between light and annotated tags until today.

**Comment by rcorre** [comment]
Thanks, I tried `--tags` as well but `airbnb_1.7.1` wasn't much more informative.

Finally found it with ` git branch -r --contains  bb0d9ee1`

> There isn't yet a 1.10 tag on branch v1-10-test

Does this mean there are no 1.10 release candidates yet?

**Pull request closed without merging**